OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -53,9 +54,17 @@ public GameInfoStatsLogic(Widget widget, World world)
 				nameLabel.GetColor = () => pp.Color.RGB;
 
 				var flag = item.Get<ImageWidget>("FACTIONFLAG");
-				flag.GetImageName = () => pp.Country.Race;
 				flag.GetImageCollection = () => "flags";
-				item.Get<LabelWidget>("FACTION").GetText = () => pp.Country.Name;
+				if (pp.World.RenderPlayer != null && pp.World.RenderPlayer.Stances[pp] == Stance.Ally)
```

Comment:
I agree.  The stats and menu panels are separate to the observer visibility selector, so should not depend on its state (i.e. RenderPlayer).